### ISSUE TITLE
Schematic

### ISSUE THREAD

**Issue by lzref** [eudes/korg-nts1-esp32#2]
Hi! Do you plan to publish the schematic for your project?

I've been struggling with NTS-1 for days and days and your resource seems to be the best analysis/documentation out there.

I'm just trying to make the STM32F103C8T6 "Blue Pill" to work with NTS-1. I was even able to compile and upload the original nts-1-customizations code to it (by changing includes and changing pins; I'm trying to use the SPI1 instead of SPI2). But when I connect everything, it simply does nothing (a MIDI device doesn't even appear on the computer connected to NTS-1). So maybe I connected things incorrectly.

You mention that you lost several days because you didn't connect all the pins properly. Maybe I got something similar.

I also have an ESP-WROOM-32 board from DOIT but I have zero documentation on it, so I don't even know what pins can be used for SPI.

All in all, I'm just desperate to find some setup that does _something_. As long as it initializes NTS-1 and gets ready to exchange SPI messages, I'm down to reverse-engineer them if the NTS-1 library doesn't work. But I don't have much experience programming SPI (other than using SPI displays, where everything's done by libraries), so this initial setup is being a big pain.

Anyway, thank you for this repo, it's been a huge help!

**Comment by eudes** [comment]
Hey! Sorry about the delay on this, I haven't been paying much attention to Github notifications.

I'm afraid I don't have access to my schematics (a scribble on a notepad which is not with me at the moment). The only documentation I have [is in the repo](https://github.com/eudes/korg-nts1-esp32#identifying-the-nts1-pins), and I'm afraid it's not very clear.

I also have [this video](https://twitter.com/eudes/status/1294697030917464066), which shows that the actual connections to IO pins on the ESP32 are very few and likely just what you can find in the tables cited above.

Just make sure that every pin in the NTS1 is connected to what the reference schematic from Korg shows (all grounded pins to ground, all 3v3 pins to 3v3).

Also, make sure that you're using the appropriate SPI IO pins on your board. Mine had only 2 sets of pins that are usable for SPI comms.

If none of this helps, check that whatever you're using as a power supply is capable of providing enough power for both the board and the NTS1. My SparkFun Thing plus connected to a laptop managed OK, but your situation might be different.

Good luck! If I find my schematics, I'll add them to the repo.

**Comment by lzref** [comment]
Thank you!

**Comment by apiel** [comment]
@lzref did you manage to come up the pin schema?

**Comment by lzref** [comment]
@apiel nope, never managed to make it work and truly confirm which pins are which

**Comment by apiel** [comment]
So i think i have a small idea base on the doc @eudes pointed out and the kicad schema.

![Image](https://github.com/user-attachments/assets/5cc32c04-3f8f-4bb8-bad3-3092526d63a2)

On the right side, there are those pins that goes to the nts1 audio board:

![Image](https://github.com/user-attachments/assets/b8d13e21-7b56-4df0-a284-8c15f3e9a523)

If we go from the top to the bottom:

- GND
- Boot
- Reset
- Tx ---> Miso
- Rx ---> Mosi
- CK ---> clock --> sck
- GND (the square one)

**Comment by lzref** [comment]
@apiel So were you able to successfully talk to the board and control it via these pins?

**Comment by apiel** [comment]
Let see next week when I receive my nts1, for the moment I don't have one :p

**Comment by lzref** [comment]
Lately I've been thinking, maybe I should just make a simple MIDI-controller for NTS-1. It's so much easier, and would even be more modular (and allow plug-and-play)
